File: fizzbuzz.cpp
#include <iostream>
#include <string>
using namespace std;

void fizzbuzz(int N)
{
    for (int i = 1; i <= N; i++) {
        string out;
        if (i % 3 == 0) {
            out = out.append("Fizz");
        }
        if (i % 5 == 0) {
            out = out.append("Buzz");
        }
        cout << out << endl;        
    }
}

int main(int argc, char *argv[]) 
{
    int N = atoi(argv[1]);
    fizzbuzz(N);
    return 0;
}

File: fizzbuzz.go
package main
import (
	"fmt"
	"os"
	"strconv"
)

func fizzbuzz(N int) {
	var out string
	i := 1
	for i <= N  {
		out = ""
		if (i % 3 == 0) {
            out += "Fizz"
        }
		if (i % 5 == 0) {
            out += "Buzz"
        }
		fmt.Println(out)
		i += 1
    }
}

func main() {
    N, _ := strconv.Atoi(os.Args[1])
    fizzbuzz(N)
}